packages/datastore/cli/src/generate-documents/createJsonSchema.ts

```diff
@@ -24,15 +21,33 @@ const getFieldParameters = (fieldName: string, type: GraphQLOutputType): any =>
 
 const getModelProperties = (model: ModelDefinition, primaryKey: string) => {
   const fieldMap = model.graphqlType.getFields();
+  const keys = Object.keys(fieldMap);
+
+  const relationships = model.relationships
+    .filter(r => !keys.includes(r.ownerField.name))
+    .map(r => {
+      const fieldOptions = getFieldParameters(
+        r.relationForeignKey!,
+        r.relationType
+      );
+      fieldOptions.relationship = r.relationType;
```

Comment:
@craicoverflow can you take look if this would work from your side. 
We actually getting model relationships and add extra fields to schema

Reply:
I'm not entirely sure yet, I think this might generate extra fields which are not part of a model. A `ModelDefinition` contains relationship information for the opposite relationship field.

For example, in a 1:M relationship between Note.comments and Comment.note, the `Note` model definition would have 2 relationships.

I think once we have a unit test in for this API, it will be easier to verify this.

Reply:
Yes you're correct @craicoverflow I had to filter out the side of the relationship that I didn't need here. But the relationship info went both ways oneToMany and manyToOne

Reply:
Is this the filtering you meant @kingsleyzissou ?

https://github.com/aerogear/offix/blob/schema-relationships/packages/datastore/cli/src/generate-documents/createJsonSchema.ts#L27

From looking at that I had thought it was filtering out the relationship information that Offix already had and did not want to apply twice.

Reply:
Yeah that's the filtering I meant @craicoverflow . As far as I can remember, we were having some issues with the comments array being included in our Datastore model and we needed to remove it.

We are only adding the foreign key information the the notes so we can filter those out based on the parent id, in this case the Task.

Reply:
See https://github.com/aerogear/offix/pull/900 - should this test be passing or am I doing something wrong?

Reply:
The test should be passing. I'm going to fix the linting and snapshot issues and then I can debug that